svelte-kit-scss/src/lib/components/IssueSearch/IssueSearchControls/IssueSearchControls.svelte

```diff
@@ -13,19 +17,53 @@
     await prefetch(option.href);
     await goto(option.href);
   };
+
+  const resolveStateFilterIssueState = (option: NavigationFilterOption): IssueState | undefined => {
+    let result = undefined;
+    if (option.label.endsWith('Closed')) {
+      result = IssueState.Closed;
+    } else if (option.label.endsWith('Open')) {
+      result = IssueState.Open;
+    }
+    return result;
+  };
 </script>
 
 <div class="issue-search-controls">
-  <DropdownMenuSelect
-    description="Sort by"
-    options={sortFilters}
-    labelAccessor={(x) => x.label}
-    checkedPredicate={(x) => x.active}
-    on:select={async ({ detail }) => await handleFilterSelect(detail)}
-  >
-    <DropdownFilterTextButton text="Sort" />
-  </DropdownMenuSelect>
+  <div class="primary">
+    <SelectButton
```

Comment:
This seems to be unnecessarily complex for what it is.

We've effectively got 2 "tabs", or links, open and closed, and one of them is currently active.

<img width="224" alt="Screenshot 2022-11-28 at 11 40 59 AM" src="https://user-images.githubusercontent.com/76821/204344831-92e918a7-6fa0-4bc7-9937-dc2a2ee155b1.png">

What you have here seems very very configurable and extendible maybe, but unnecessarily so.

I think you maybe need to take a step back here, and just have the 2 links, Open and Closed, be always visible here with their Icons hardcoded, and use the state filters to provide the 2 links with the `label`, the `href` and whether or not they have an `active` class applied.

Reply:
Yeah, but as a showcase, for someone to look at and understand, "oh this is what a SvelteKit app would look like", this doesn't fulfil the need.

This is hard for me to understand, and might give someone the impression that SvelteKit "requires" or demands code like this for very simple things, which it does not.

I would prefer anyone unfamiliar with Svelte or SvelteKit be able to look at this code and immediately recognize what part of the app it is from and how it works.

Reply:
It is the fact that this:

<img width="224" alt="Screenshot 2022-11-28 at 11 40 59 AM" src="https://user-images.githubusercontent.com/76821/204344831-92e918a7-6fa0-4bc7-9937-dc2a2ee155b1.png">

Which is 2 links, that are always showing and may have some "active" styles applied to one of them, it looks like it should be coded something like:

```svelte
<script lang="ts">
  export let stateFilters: NavigationFilterOption[];
  $: [openFilter, closedFilter] = stateFilters;
  // ...
</script>

<div class="primary">
  <a href={openFilter.href} active={openFilter.active}><Icon type="open">{openFilter.label}</a>
  <a href={closedFilter.href} active={closedFilter.active}><Icon type="closed">{closedFilter.label}</a>
</div>
```

...but ends up being this much code:

```
<script lang="ts">
  import { goto, prefetch } from '$app/navigation';
  import SelectButton from '$lib/components/shared/buttons/SelectButton.svelte';
  import DropdownMenuSelect from '$lib/components/shared/Dropdown/DropdownMenuSelect/DropdownMenuSelect.svelte';
  import { IssueState } from '$lib/interfaces';
  import IssueStateIcon from '../IssueSearchListItem/IssueStateIcon.svelte';
  import type { NavigationFilterOption } from './models';

  export let stateFilters: NavigationFilterOption[];

  const handleFilterSelect = async (option: NavigationFilterOption): Promise<void> => {
    if (option.active) {
      return;
    }
    await prefetch(option.href);
    await goto(option.href);
  };

  $: console.log('stateFilters', stateFilters);

  const resolveStateFilterIssueState = (option: NavigationFilterOption): IssueState | undefined => {
    let result = undefined;
    if (option.label.endsWith('Closed')) {
      result = IssueState.Closed;
    } else if (option.label.endsWith('Open')) {
      result = IssueState.Open;
    }
    return result;
  };
</script>
//...
<div class="primary">
    <SelectButton
      options={stateFilters}
      labelAccessor={(x) => x.label}
      checkedPredicate={(x) => x.active}
      on:select={async ({ detail }) => await handleFilterSelect(detail)}
    >
      <svelte:fragment slot="option-icon-left" let:option>
        {#each [resolveStateFilterIssueState(option)] as state}
          {#if state}
            <IssueStateIcon {state} />
          {/if}
        {/each}
      </svelte:fragment>
    </SelectButton>
</div>
```
```
<script lang="ts">
  import ButtonText from '$lib/components/shared/buttons/ButtonText.svelte';
  import { createEventDispatcher } from 'svelte';

  type TOption = $$Generic;
  export let options: TOption[];
  export let labelAccessor = (option: TOption) => String(option);
  export let checkedPredicate: (option: TOption) => boolean = () => false;

  const dispatch = createEventDispatcher<{ select: TOption }>();

  const handleOptionClick = (option: TOption): void => {
    dispatch('select', option);
  };

  $: console.log($$slots);
</script>

<div class="select-button">
  {#each options as option}
    <div
      class="button-wrapper"
      class:active={checkedPredicate(option)}
      on:click={() => handleOptionClick(option)}
      on:keypress={() => handleOptionClick(option)}
    >
      <ButtonText text={labelAccessor(option)}>
        <svelte:fragment slot="left">
          <slot name="option-icon-left" {option} />
        </svelte:fragment>
        <svelte:fragment slot="right">
          <slot name="option-icon-right" {option} />
        </svelte:fragment>
      </ButtonText>
    </div>
  {/each}
</div>
```

...and now it includes needing to understand fragments, slots, event handling, createEventDispatcher, the `$$Generic` syntax, what `prefetch` and `goto` are and do.

I understand wanting to not "hard code" the component, but this is just too far in the other direction and would only serve to confuse or baffle someone coming to learn from our showcase

Reply:
> Should I make it less reusable?

Yes.  There is a balance between readable and reusable